enacts/calc.py

```diff
@@ -121,6 +126,113 @@ def synthesize_enacts(variable, time_res):
     var_name = variable
     return xrds[var_name]
 
+
+def sql2GeoJSON(shapes_sql, db_config):
+    """ Form a GeoJSON dict from sql request to a database
+
+    Parameters
+    ----------
+    shapes_sql: str
+        sql request
+    db_config: dict
+        dictionary with host, port, user and dbname information
+    
+    Returns
+    -------
+    features: dict
+        dictionary with features as key and GeoJSON of shapes_sql as value
+
+    See Also
+    --------
+    sql2geom, geom2GeoJSON
+
+    Examples
+    --------
+    shapes_sql: select id_1 as key, name_1 as label,
+        ST_AsBinary(the_geom) as the_geom from sen_adm1
+    db_config:
+        host: postgres
+        port: 5432
+        user: ingrid
+        dbname: iridb
+    """
+    return geom2GeoJSON(sql2geom(shapes_sql, db_config))
+
+
+def geom2GeoJSON(df):
+    """ Form a GeoJSON dict from a geometric object
+
+    Parameters
+    ----------
+    df: geometric object
+        shapely geometric object
+    
+    Returns
+    -------
+    features: dict
+        dictionary with features as key and GeoJSON of `geom` as value
+
+    See Also
+    --------
+    sql2geom, shapely.MultiPolygon, shapely.geometry.mapping
+    """
+    df["the_geom"] = df["the_geom"].apply(
+        lambda x: x if isinstance(x, MultiPolygon) else MultiPolygon([x])
+    )
+    shapes = df["the_geom"].apply(shapely.geometry.mapping)
+    for i in df.index: #this adds the district layer as a label in the dict
+        shapes[i]['label'] = df['label'][i]
+    return {"features": shapes}
+
+
+def sql2geom(shapes_sql, db_config):
+    """ Form a geometric object from sql query to a database
+
+    Parameters
+    ----------
+    shapes_sql: str
+        sql query
+    db_config: dict
+        dictionary with host, port, user and dbname information
+    
+    Returns
+    -------
+    df : pandas.DataFrame
+        a pd.DF including "the_geom" : a shapely geometric object
```

Comment:
```
a pd.DataFrame with columns "label" (dtype=string), "key" (string or int depending on the table), and "the_geom" (shapely.Geometry)
```
FWIW the fact that `key` is sometimes a string and sometimes a number is a bad design that tends to lead to bugs. But that's not at issue right now.